src/include/miopen/pad_constant/problem_description.hpp

```
@@ -0,0 +1,209 @@
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include "miopen/errors.hpp"
+#include "miopen/names.hpp"
+#include "miopen/problem_description_base.hpp"
+#include "miopen/tensor.hpp"
+#include <cstddef>
+#include <miopen/miopen.h>
+
+namespace miopen {
+namespace pad_constant_fwd {
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const TensorDescriptor& xDesc_,
+                       const TensorDescriptor& yDesc_,
+                       const size_t* padding_,
+                       const int padding_size_ = 0)
+        : xDesc(xDesc_), yDesc(yDesc_), padding(padding_), padding_size(padding_size_)
+    {
+        // Consistency checks
+        if(!IsPaddingValid())
+            MIOPEN_THROW("Padding is not valid");
+        if(!IsSameShape())
+            MIOPEN_THROW("Tensors do not have the same shapes");
+        if(!IsSameType())
+            MIOPEN_THROW("Tensor values do not have the same type");
+    }
+
+    const TensorDescriptor& GetXDesc() const { return xDesc; }
+    const TensorDescriptor& GetYDesc() const { return yDesc; }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+    bool IsSameType() const
+    {
+        if(xDesc.GetType() == yDesc.GetType())
+            return true;
+        return false;
+    }
+
+    bool IsSameShape() const
+    {
+        auto xSize = xDesc.GetSize();
+        auto ySize = yDesc.GetSize();
+        if(xSize == ySize)
+            return true;
+        return false;
+    }
+
+    bool IsContiguous() const { return xDesc.IsContiguous() && yDesc.IsContiguous(); }
+
+    bool IsImprovementOverROCm() const
+    {
+        if(IsContiguous())
+            // No contiguous case is faster
+            return false;
+        else
+            // Slower if n is padded (at all)
+            return padding[0] == 0 && padding[1] == 0;
+    }
+
+    bool IsPaddingValid() const
```

If the input value is not valid, throw rather than return false.

---

I've added that check to the initializer which would also call `MIOPEN_THROW`, so I don't think that would be necessary here. I can add it if needed